test/modules/CloudToDeviceMessageTester/CloudToDeviceMessageTesterMetadata.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace CloudToDeviceMessageTester
+{
+    using System;
+    using Microsoft.Azure.Devices.Client;
+
+    public class CloudToDeviceMessageTesterMetadata
```

Comment:
First, why not just use Settings class?
Second, if you want to have different settings for sender and receiver.  I would suggest:
1. Create 2 classes SenderTestSettings and ReceiverTestSettings, both extends from TestSettings.
2. In TestSettings class, it contains properties in shared metadata.
3. In SenderTestSettings, it contains properties in SenderMetadata; and Create method.  Same for ReceiverTestSettings.
4. In Program.cs, you just need to create corresponding Settings class inside the TestMode if-block.

This will make the program only get setting parameters only related to test mode; and devs only need to look at 1 settings file depending on which test mode they are interested.